modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java

```diff
@@ -554,6 +567,36 @@ private Filter _getORFilter(Filter leftFilter, Filter rightFilter) {
 		return booleanFilter;
 	}
 
+	private Object _getSUB(Object left, Object right) {
+		if ((left == null) || (right == null)) {
+			throw new UnsupportedOperationException(
+				"Unsupported operator SUB with null operands");
+		}
+
+		try {
+			Date date = (Date)_format.parseObject(left.toString());
```

Comment:
You need to check that the left and the right are the the right values.

It only work if the left is a String and it has a valid date/time format.

So would be better that you only run the parser (a costly operation) if the left and right is a String, then the exception catch are OK.

But for example if you are running the sub operation in a boolean operation, a different erros should be thrown.

Reply:
done!